beeline/src/java/org/apache/hive/beeline/JSONOutputFormat.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This source file is based on code taken from SQLLine 1.9
+ * See SQLLine notice in LICENSE
+ */
+package org.apache.hive.beeline;
+
+import java.sql.SQLException;
+import java.sql.Types;
+
+import com.fasterxml.jackson.core.io.JsonStringEncoder;
+
+/**
+ * OutputFormat for standard JSON format.
+ *
+ */ 
+public class JSONOutputFormat extends AbstractOutputFormat {
+  private final BeeLine beeLine;
+  private int[] columnTypes;
+  private JsonStringEncoder jse = JsonStringEncoder.getInstance();
```

Comment:
Take a look at this:

https://fasterxml.github.io/jackson-core/javadoc/2.8/com/fasterxml/jackson/core/JsonGenerator.html

Please use the Jackson JSON generator to... generate JSON. :)